src/components/Trade/Transaction/Transaction.jsx

```diff
@@ -2,99 +2,108 @@ import React, { Component } from 'react';
 import Modal from 'react-responsive-modal';
 import Select from 'react-select';
 import Button from '../../Button/Button';
+import { transaction } from '../../../api/api';
+import { withAlert } from 'react-alert'
 import './Transaction.css';
 
 const tradeOptions = [
-	{ value: 'buy', label: 'Buy' },
-	{ value: 'sell', label: 'Sell' }
+  { value: 'buy', label: 'Buy' },
+  { value: 'sell', label: 'Sell' }
 ]
 
 
 class Transaction extends Component {
 
-	constructor(props) {
-		super(props);
+  constructor(props) {
+    super(props);
 
-		this.state = {
-			selectedOption: { value: 'buy', label: 'Buy' },
-			buttonDisabled: true,
-			amount: ""
-		}
-	}
+    this.state = {
+      selectedOption: { value: 'buy', label: 'Buy' },
+      buttonDisabled: true,
+      amount: ""
+    }
+  }
 
-	handleSelectChange = (selectedOption) => {
-		this.setState({ selectedOption });
-	}
+  handleSelectChange = (selectedOption) => {
+    this.setState({ selectedOption });
+  }
 
-	onChange = (event) => {
-		var newState = this.state;
+  onChange = (event) => {
+    var newState = this.state;
```

Comment:
It's considered bad practice to directly modify state like this, better to make a copy Object.assign({}, this.state) or {...this.state}

Reply:
Good catch!